File: api/src/main/java/com/hirrao/javaee/service/DrugService.java
package com.hirrao.javaee.service;

import com.hirrao.javaee.entity.DrugManageResult;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface DrugService {
    void insertDrug(long drugId,long uid,String drugName,String frequency,String unit,Float dosage,int isActive);
//    List<DrugManageResult> getAllDrugInfo(long uid);

    List<DrugManageResult> getPageDrugInfo(long uid,int curPage,int size);

    List<DrugManageResult> getPageDrugInfoByDrugName(long uid,int curPage,int size,String drugName);

    int getPageDrugInfoTotal(long uid);

    int getPageDrugInfoTotalByDrugName(long uid,String drugName);

    List<DrugManageResult> getPageDrugAlertInfo(long uid,int curPage,int size);

    List<DrugManageResult> getPageDrugAlertInfoByDrugName(long uid,int curPage,int size,String drugName);

    int getPageDrugAlertInfoTotal(long uid);

    int getPageDrugAlertInfoTotalByDrugName(long uid,String drugName);

    void updateDrugIsActiveById(long drugId,long uid,int isActive);

    void deleteDrugById(long uid,long drugId);
}

File: api/src/main/kotlin/cn/hirrao/javaee/service/DrugService.kt
package cn.hirrao.javaee.service

import cn.hirrao.javaee.entity.DrugManageResult
import org.springframework.stereotype.Service

@Service
interface DrugService {
    fun insertDrug(
        drugId: Long,
        uid: Int,
        drugName: String?,
        frequency: String?,
        unit: String?,
        dosage: Float?,
        isActive: Int
    )

    //    List<DrugManageResult> getAllDrugInfo(long uid);
    fun getPageDrugInfo(uid: Int, curPage: Int, size: Int): List<DrugManageResult?>?

    fun getPageDrugInfoByDrugName(uid: Int, curPage: Int, size: Int, drugName: String?): List<DrugManageResult?>?

    fun getPageDrugInfoTotal(uid: Int): Int

    fun getPageDrugInfoTotalByDrugName(uid: Int, drugName: String?): Int

    fun getPageDrugAlertInfo(uid: Int, curPage: Int, size: Int): List<DrugManageResult?>?

    fun getPageDrugAlertInfoByDrugName(uid: Int, curPage: Int, size: Int, drugName: String?): List<DrugManageResult?>?

    fun getPageDrugAlertInfoTotal(uid: Int): Int

    fun getPageDrugAlertInfoTotalByDrugName(uid: Int, drugName: String?): Int

    fun updateDrugIsActiveById(drugId: Long, uid: Int, isActive: Int)

    fun deleteDrugById(uid: Int, drugId: Long)
}
